src/stapi_fastapi/routers/product_router.py

```diff
@@ -224,7 +221,7 @@ async def create_order(
             request,
         ):
             case Success(order):
-                self.root_router.add_order_links(order, request)
+                order.links.extend(self.root_router.order_links(order, request))
```

Comment:
improves the semantics of the function to make it more functional -- e.g., it doesn't mutate it's parameters, and instead just returns an object that the caller can then use for mutation